src/Neo.SmartContract.Framework/Attributes/NoReentrantAttribute.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2015-2022 The Neo Project.
+// 
+// The Neo.SmartContract.Framework is free software distributed under the MIT 
+// software license, see the accompanying file LICENSE in the main directory 
+// of the project or http://www.opensource.org/licenses/mit-license.php 
+// for more details.
+// 
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System;
+using Neo.SmartContract.Framework.Services;
+
+namespace Neo.SmartContract.Framework.Attributes
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public class NoReentrantAttribute : ModifierAttribute
+    {
+        private readonly StorageMap _context;
+        private readonly string _key;
+
+        public NoReentrantAttribute(byte prefix = 0xFF, string key = "noReentrant")
```

Comment:
```suggestion
        public NoReentrantAttribute(byte prefix = 0xFF, [CallerMemberName]string key = "noReentrant")
```

@shargon Can you test this?

Reply:
Yes, but it will define only the default value? 
It's good to have the same value for `sell` and `buy` because you should not be able to reentrant between different methods by default

Reply:
> It's good to have the same value for sell and buy because you should not be able to reentrant between different methods by default

If you provide an argument explicitly, the value you provide will be used.

My concern is that the `CallerMemberNameAttribute` doesn't work properly in the contract. Need to actually test it in the contract.

Reply:
> Need to actually test it in the contract.

I tested and it looks good. If someone else can test it should be good, but seems that it works.